### DISCUSSION TITLE
(aws-cdk-lib/aws-rds): How to specify zones or set Multi-AZ with DatabaseCluster in AWS CDK

### DISCUSSION THREAD

**Discussion by pgilewski** [aws/aws-cdk#27033] [Q&A]
I have this aws cdk in typescript function which deploy database cluster of serverlessV2 instances, I cannot find a way to deploy my reader in different zone than writer.

How do I deploy ClusterInstances in Multi Availability Zones using Aws cdk?

Or how can I specify zones of database instances using Aws cdk?

```typescript
  createDatabaseCluster(id: string, props: DatabaseProps) {
    const cluster = new rds.DatabaseCluster(this, id, {
      engine: rds.DatabaseClusterEngine.auroraPostgres({
        version: rds.AuroraPostgresEngineVersion.VER_13_6,
      }),
      defaultDatabaseName: props.databaseName,
      vpc: this.vpc,
      vpcSubnets: this.vpc.selectSubnets({
        subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
      }),
      writer: rds.ClusterInstance.serverlessV2('writer', {
        enablePerformanceInsights: false,
        autoMinorVersionUpgrade: true,
      }),
      readers: [
        rds.ClusterInstance.serverlessV2('reader', {
          scaleWithWriter: false,
          enablePerformanceInsights: false,
          autoMinorVersionUpgrade: true,
        }),
      ],
      serverlessV2MinCapacity: 0.5,
      serverlessV2MaxCapacity: 2,
      credentials: rds.Credentials.fromGeneratedSecret(props.username, {
        secretName: `/${props.project.name}/database/credentials`,
      }),
      storageEncryptionKey: this.storageEncryptionKey,
      backup: {
        retention: cdk.Duration.days(3),
      },
    });

    return cluster;
  }
```

**Comment by rantoniuk** [comment]
Aurora Serverless v2 [handles high-availability and data redundancy automatically](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.AuroraHighAvailability.html). If the cluster spans across subnets in multiple AZs, the readers should be created by default in another AZ than the writer and promoted in case of failure of the primary instance.
Note you have to set `scaleWithWriter: true` for at least one reader as the CDK suggests.
